internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -867,6 +849,10 @@ func (s *SiteConfigLinuxWebAppSlot) ExpandForUpdate(metadata sdk.ResourceMetaDat
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.auto_heal_setting") {
+		expanded.AutoHealEnabled = pointer.To(false)
```

Comment:
as above, if there's no change we're automatically turning auto-heal off, so we should only set it here, not at the top of the function?